File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cpp
#include <vector>

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0, right = numbers.size() - 1;
        while (left < right) {
            int sum = numbers[left] + numbers[right];
            if (sum == target)
                return vector<int>{left + 1, right + 1};
            else if (sum > target) 
                right -= 1;
            else
                left += 1;
        }
        return vector<int>();
    }
};

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n = len(numbers)
        left, right = 0, n - 1
        
        while left < right:
            sm = numbers[left] + numbers[right]
            if sm == target:
                return [left + 1, right + 1]
            left, right = (left, right - 1) if sm > target else (left + 1, right)
        
        return None